### ISSUE TITLE
No fields with distinct gives NPE

### ISSUE THREAD

**Issue by santanusinha** [Flipkart/foxtrot#111]
Query:
```
{
    "opcode": "count",
    "filters": [
        {
            "operator": "last",
            "field": "_timestamp",
            "currentTime": 1452265792522,
            "duration": "1 day",
            "temporal": true
        }
    ],
    "table": "testapp",
    "distinct": true
}
```

StackTrace:
Caused by: ! java.lang.NullPointerException: null
! at com.flipkart.foxtrot.core.querystore.actions.Utils.sanitizeFieldForAggregation(Utils.java:36) ~[server.jar:na]
! at com.flipkart.foxtrot.core.querystore.actions.CountAction.execute(CountAction.java:87) ~[server.jar:na]

Proper behaviour: Generate an error as the query is wrong, field always needs to be present when distinct is specified.

**Comment by r0goyal** [comment]
@santanusinha Working on #68 which should fix this one as well

**Comment by r0goyal** [comment]
Fixed by #140
